docs/book/src/development/development.md

```diff
@@ -342,6 +342,76 @@ $ sudo ln -s /run/user/$(id -u)/podman/podman.sock /var/run/docker.sock
 [hack-ci-gce-project]: https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/main/hack/ci/gce-project.sh
 [hack-ci-openstack]: https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/main/hack/ci/openstack.sh
 
+## Cluster Creation with ClusterClass and Tilt
+
+This section provides a guide to create a Kubernetes cluster using ClusterClass in the development environment. It covers the process of setting up environment variables, creating a Kind cluster, configuring a secret, applying ClusterClass, deploying the cluster, monitoring its provisioning, obtaining Kubeconfig, and applying CNI to the cluster.
+
+#### Setting Environment Variables
+
+Define necessary environment variables relevant to your cluster and infrastructure. This includes setting variables for SSH key names and Kubernetes version.
+
+```bash
+#### Set the name for the SSH keypair used for instances in OpenStack
+export OPENSTACK_SSH_KEY_NAME="your_ssh_key_name_here"
+#### Set the desired Kubernetes version for your cluster
+export KUBERNETES_VERSION="v1.28.5"
+```
+
+#### Creating a Kind Cluster
+
+Create a Kind cluster. This process involves the initialization of Cluster API providers for OpenStack.
+
+```bash
+kind create cluster
+export CLUSTER_TOPOLOGY=true
+clusterctl init --infrastructure openstack
+```
+
+#### Secret Configuration
+
+Apply a Kubernetes secret that your cluster configuration will reference. Ensure the secret's name aligns with the required format in your configuration.
+
+#### Applying ClusterClass and Cluster Configuration
+
+Substitute variables in your ClusterClass configuration file and apply it to set up the cluster class.
+
+```bash
+envsubst < templates/clusterclass-dev-test.yaml | kubectl apply -f -
+envsubst < templates/cluster-template-development.yaml | kubectl apply -f -
```

Comment:
Since we used `clusterctl` above, I would say we should use it also here. It will complain if you forgot some variables and has flags for number of replicas which can be nice.